x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/SumDoubleAggregator.java

```diff
@@ -57,8 +57,11 @@ public static void evaluateIntermediate(SumState state, DriverContext driverCont
     }
 
     public static Block evaluateFinal(SumState state, DriverContext driverContext) {
-        double result = state.value();
-        return driverContext.blockFactory().newConstantDoubleBlockWith(result, 1);
+        double value = state.value();
+        if (Double.isFinite(value) == false) {
+            return driverContext.blockFactory().newConstantNullBlock(1);
```

Comment:
I guess we're currently auto-converting non-finite doubles to nulls somewhere, need some opinion on this one.

The scalar functions tests check that they don't return a non-finite double, and so we're doing the same in aggs.

Reply:
We usually emit a WARNING when we do it. It'd be nice if we could do that here too.

Does this trigger because we add together enough big numbers to shift us to +Infinity or something?